python/examples/grpc_impl_example/fit_a_line/test_asyn_client.py

```diff
@@ -42,9 +41,16 @@ def call_back(call_future, data):
 
 task_count = 0
 for data in test_reader():
-    future = client.predict(feed={"x": data[0][0]}, fetch=["price"], asyn=True)
-    task_count += 1
-    future.add_done_callback(functools.partial(call_back, data=data))
+    try:
+        future = client.predict(
+            feed={"x": data[0][0]}, fetch=["price"], asyn=True)
+    except grpc.RpcError as e:
```

Comment:
do we need to expose the grpc exception here?

Reply:
fix it: catch grpc exception inside Client and return status_code in fetch_map.